### PR TITLE
Fixes #18908: getLocalization calls still existing in dijit despite the method being deprecated

### PR DESCRIPTION
**Pull request by pottedmeat**

### CODE REVIEW

**Comment by wkeese** [comment]
The calls to `getLocalization()` are intentional.  Your change breaks backwards compatibility, specifically `dojo.require()` (i.e. non-AMD modules).

**Comment by dylans** [comment]
@wkeese Without these changes, support for [CSP](https://developer.mozilla.org/en-US/docs/Web/Security/CSP) when i18n is used is not possible, which is becoming increasingly important for our Dojo 1.x users. So what do you propose? Could we create a branch in the assignment based on whether they are using the async or sync loader? I'm really regretting the decision to not drop non-AMD support a while back. :(

**Comment by wkeese** [comment]
Yah, I don't know if there's a silver bullet where you can support CSP without breaking our promise of backwards compatibility.  You should also double check that change breaks non-AMD files.

I don't have a personal stake in dropping non-AMD support, but it would be the biggest violation of our backwards-compatibility promise ever, and I know people have screamed bloody murder over much more trivial changes.

In any case, assuming this change breaks non-AMD files, then before checking it in, all the non-AMD files in the dojo source that depend on dijit would need to be converted to AMD.   That's 95 files in dijit itself and some subset of the 970 non-AMD files in dojox.

File: ColorPalette.js [hunk 1]
```diff
@@ -5,12 +5,11 @@ define([
 	"./_TemplatedMixin",
 	"./_PaletteMixin",
 	"./hccss", // has("highcontrast")
-	"dojo/i18n", // i18n.getLocalization
```
Comment (dylans) [inline ColorPalette.js]: To address @wkeese's point about still needing to support non-AMD code, I'm thinking we could do this:

"dojo/has!csp-restrictions?dojo/i18n!dojo/nls/colors:dojo/i18n" // i18n or i18n.getLocalization

File: ColorPalette.js [hunk 2]
```diff
@@ -84,7 +83,7 @@ define([
 			// Creates <img> nodes in each cell of the template.
 			this._preparePalette(
 				this._palettes[this.palette],
-				i18n.getLocalization("dojo", "colors", this.lang));
```
Comment (dylans) [inline ColorPalette.js]: And then prior to the call to _preparePalette:

var _i18n;
if(has("csp-restrictions")){
	_i18n = i18n.getLocalization("dojo", "colors", this.lang));
}else{
	_i18n = i18n;
}

File: ColorPalette.js (hunk 1 above)
Comment (wkeese) [inline ColorPalette.js]: Oh that's a good idea.   I like the idea of doing it whenever we are using the async loader.  Can you branch based on that `async` flag that we usually set when we include dojo.js?

```
<script type="text/javascript" src="dojo.js"
	data-dojo-config="async: true"></script>
```

Hopefully that's exposed as a has() flag.

Comment (dylans) [inline ColorPalette.js]: I believe that has flag would be config-async ?

Comment (pottedmeat) [inline ColorPalette.js]: It looks like there's `has("dojo-loader") and has("dojo-async-loader")`.

**Comment by pottedmeat** [comment]
Closing to submit an alternative fix.

**Comment by paulrutter** [comment]
@pottedmeat I had a monkeypatch in place for years, which fixed this issue as well (always thought that i was the only one having this race condition!).
Now, i want to remove my monkey patches, but i notice that i18n.getLocalization() is still being used in the dojox package (see _BusyButtonMixin for example).
Do i need to file a separate issue for this?

**Pull request closed without merging**